sdk/python/kfp/compiler/compiler.py

```diff
@@ -48,7 +48,7 @@ def my_pipeline(a: int, b: str = 'default value'):
 
     def compile(
         self,
-        pipeline_func: base_component.BaseComponent,
+        pipeline_func: Union[base_component.BaseComponent, Callable],
```

Comment:
I might be missing something, when would the compile method get a callable that's not a subclass of `BaseComponent`?

Reply:
You are correct, the original annotation is correct. Discussed offline. Will revert.

Reply:
Reversion PR: https://github.com/kubeflow/pipelines/pull/9095